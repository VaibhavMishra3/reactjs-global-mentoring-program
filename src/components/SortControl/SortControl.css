.sort-control {
  display: flex;
  align-items: center;
  color: #ffffff;
  gap: 10px;
  font-size: 16px;
}

.sort-control label {
  margin-right: 10px;
  font-weight: bold;
}

.sort-control select {
  padding: 5px;
  border-radius: 4px;
  background-color: #424242;
  color: #ffffff;
  border: 1px solid #555555;
  outline: none;
}

.sort-control select option {
  background-color: #232323;
  color: #ffffff;
}
