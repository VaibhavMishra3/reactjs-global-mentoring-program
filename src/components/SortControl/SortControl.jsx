import React, { useState } from 'react';
import { PropTypes } from 'prop-types';
import { Form } from 'react-bootstrap';

const SortControl = ({ sortOptions, defaultSort, handleChange }) => {
  const [sort, setSort] = useState(defaultSort);

  const handleSortChange = (event) => {
    setSort(event.target.value);
    handleChange(event.target.value);
  }

  return (
    <div className='d-flex align-items-center'>
      <div>
        <label>Sort by</label>
      </div>
      <div className='m-1'>
        <Form.Select data-testid='sort-select' value={sort} onChange={handleSortChange}>
          {sortOptions.map((item) => (
            <option key={item.value} value={item.value}>
              {item.name}
            </option>
          ))}
        </Form.Select>
      </div>
    </div>
  )
}

SortControl.propTypes = {
  sortOptions: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired
  })).isRequired,
  defaultSort: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired
};

export default SortControl;
