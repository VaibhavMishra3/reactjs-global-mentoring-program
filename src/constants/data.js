export const GENRES = [
  'Action',
  'Adventure',
  'Comedy',
  'Crime',
  'Fantasy',
  'History',
  'Science Fiction',
  'Thriller',
  'Western'
];

export const SORT_OPTIONS = [
  {
    name: 'Title',
    value: 'title'
  },
  {
    name: 'Release date',
    value: 'release_date'
  }
];

export const DEFAULT_SEARCH = '';
export const DEFAULT_GENRES = [];
export const DEFAULT_SORT = SORT_OPTIONS[0].value;

export const SEARCH_QUERY_PARAM = 'search';
export const GENRES_QUERY_PARAM = 'genres';
export const SORT_QUERY_PARAM = 'sort';
