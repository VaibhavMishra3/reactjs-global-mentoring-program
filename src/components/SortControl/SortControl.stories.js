import SortControl from './SortControl';
import { SORT_OPTIONS } from '../../constants/data.js';

export default {
  component: SortControl
};

export const Default = {
  args: {
    sortOptions: SORT_OPTIONS,
    defaultSort: SORT_OPTIONS[1].value,
    handleChange: (event) => console.log(`sort by: ${event.target.value}`)
  },
  parameters: {
  }
};
