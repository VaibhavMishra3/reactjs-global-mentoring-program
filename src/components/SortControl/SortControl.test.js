import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event';

import SortControl from './SortControl';
import { SORT_OPTIONS } from '../../constants/data.js';

describe('SortControl', () => {

  afterEach(cleanup);

  it('should match snapshot', () => {
  const sortControl = renderer
    .create(
    <SortControl
      sortOptions={SORT_OPTIONS}
      defaultSort={SORT_OPTIONS[1].value}
      handleChange={console.log}
    />
    )
    .toJSON();

  expect(sortControl).toMatchSnapshot();
  });

  it('render correctly', () => {
  render(
    <SortControl
    sortOptions={SORT_OPTIONS}
    defaultSort={SORT_OPTIONS[1].value}
    handleChange={console.log}
    />
  );

  expect(screen.getByRole('combobox')).toHaveValue('release_date');
  expect(screen.getByRole('option', { name: 'Title' }).selected).toBe(false);
  expect(screen.getByRole('option', { name: 'Release date' }).selected).toBe(true);
  });

  it('should select value', async () => {
  const { user } = {
    user: userEvent.setup(),
    ...render(
    <SortControl
      sortOptions={SORT_OPTIONS}
      defaultSort={SORT_OPTIONS[1].value}
      handleChange={console.log}
    />
    )
  };

  expect(screen.getByRole('combobox')).toHaveValue('release_date');
  expect(screen.getByRole('option', { name: 'Title' }).selected).toBe(false);
  expect(screen.getByRole('option', { name: 'Release date' }).selected).toBe(true);

  await waitFor(async () => {
    await user.selectOptions(screen.getByRole('combobox'), 'title');
  });

  expect(screen.getByRole('combobox')).toHaveValue('title');
  expect(screen.getByRole('option', { name: 'Title' }).selected).toBe(true);
  expect(screen.getByRole('option', { name: 'Release date' }).selected).toBe(false);
  });

});
