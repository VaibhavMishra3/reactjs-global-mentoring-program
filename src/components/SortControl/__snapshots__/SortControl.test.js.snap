// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortControl should match snapshot 1`] = `
<div
  className="d-flex align-items-center"
>
  <div>
    <label>
      Sort by
    </label>
  </div>
  <div
    className="m-1"
  >
    <select
      className="form-select"
      data-testid="sort-select"
      onChange={[Function]}
      value="release_date"
    >
      <option
        value="title"
      >
        Title
      </option>
      <option
        value="release_date"
      >
        Release date
      </option>
    </select>
  </div>
</div>
`;
